warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexRangeIteratorJexl.java

```diff
@@ -109,60 +117,90 @@ public String toString() {
     
     @Override
     protected List<Range> buildBoundingFiRanges(Text rowId, Text fiName, Text fieldValue) {
-        Key startKey = null;
-        Key endKey = null;
-        // construct new range
-        
+        if (subRanges != null && !subRanges.isEmpty()) {
+            List<Range> ranges = new ArrayList<>();
+            if (isNegated()) {
+                Key startFi = new Key(rowId, fiName);
+                Key endFi = new Key(rowId, new Text(fiName.toString() + '\0'));
+                
+                Key startKey = null;
+                for (Range subRange : subRanges) {
+                    Range range;
+                    if (subRange == subRanges.first()) {
+                        range = new Range(startFi, true, createLowerBoundKey(rowId, fiName, subRange.getStartKey().getRow(), subRange.isStartKeyInclusive()),
+                                        true);
+                        startKey = createUpperBoundKey(rowId, fiName, subRange.getEndKey().getRow(), subRange.isEndKeyInclusive());
+                    } else if (subRange == subRanges.last()) {
+                        range = new Range(startKey, true, endFi, true);
+                    } else {
+                        range = new Range(startKey, true, createLowerBoundKey(rowId, fiName, subRange.getStartKey().getRow(), subRange.isStartKeyInclusive()),
```

Comment:
If we initialized startKey to startFi, I think this if/else block could be simplified to just `if (subRange == subRanges.last()) /* final fi range */ else /* intermediate range */`

Reply:
That's a great suggestion.  Thanks, dude!